### DISCUSSION TITLE
Modifying Gitlab token recreate Dashboard issue and MR

### DISCUSSION THREAD

**Discussion by philippe-granet** [renovatebot/renovate#18431] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.238.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.3.4

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

For security reasons, I have to change Gitlab token (and user) used by Renovate.  
When I have done that, Renovate :
* create a new Dashboard issue
* recreate already closed MR
* for already open MR, an error is logged (`Another open merge request already exists for this source branch`)

### Relevant debug logs

<details><summary>Logs</summary>

```
2022-10-20T10:11:28.589Z DEBUG: Pull request creation error (repository=<repo>, baseBranch=master, branch=renovate/npm-8.x)
       "err": {
         "name": "HTTPError",
         "code": "ERR_NON_2XX_3XX_RESPONSE",
         "timings": {
           "start": 1666260688450,
           "socket": 1666260688451,
           "lookup": 1666260688454,
           "connect": 1666260688455,
           "secureConnect": 1666260688457,
           "upload": 1666260688457,
           "response": 1666260688584,
           "end": 1666260688584,
           "phases": {
             "wait": 1,
             "dns": 3,
             "tcp": 1,
             "tls": 2,
             "request": 0,
             "firstByte": 127,
             "download": 0,
             "total": 134
           }
         },
         "message": "Response code 409 (Conflict)",
         "stack": "HTTPError: Response code 409 (Conflict)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/32.238.1 (https://github.com/renovatebot/renovate)",
             "private-token": "**redacted**",
             "accept": "application/json",
             "content-type": "application/json",
             "content-length": "3263",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://hosted.gitlab.server/api/v4/projects/<repo>/merge_requests",
           "hostType": "gitlab",
           "username": "",
           "password": "",
           "method": "POST",
           "http2": true
         },
         "response": {
           "statusCode": 409,
           "statusMessage": "Conflict",
           "body": {
             "message": [
               "Another open merge request already exists for this source branch: !154"
             ]
           },
           "headers": {
             "date": "Thu, 20 Oct 2022 10:11:28 GMT",
             "server": "Apache",
             "x-frame-options": "SAMEORIGIN, SAMEORIGIN, SAMEORIGIN",
             "strict-transport-security": "max-age=15768000; includeSubDomains; preload",
             "cache-control": "no-cache",
             "content-type": "application/json",
             "vary": "Origin",
             "x-content-type-options": "nosniff",
             "x-request-id": "01GFTE0QJ93WHJ876MR7BF2FHZ",
             "x-runtime": "0.123208",
             "content-length": "86",
             "access-control-allow-methods": "*",
             "access-control-allow-origin": "*",
             "access-control-allow-headers": "*",
             "x-xss-protection": "1; mode=block",
             "referrer-policy": "strict-origin-when-cross-origin",
             "connection": "close"
           },
           "httpVersion": "1.1"
         }
       }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by PhilipAbed** [comment, marked as answer]
@philippe-granet that is expected, renovate filters Pull Requests by the User that creates them.
if you change the user, then it will cause such issues,
make sure you use only one user for renovate to open/close PRs on the repo.

**Reply by philippe-granet** [reply]
@PhilipAbed Oh, I just see this option:
https://docs.renovatebot.com/configuration-options/#ignoreprauthor

>This is usually needed if someone needs to migrate bot accounts, including from hosted app to self-hosted.
>If ignorePrAuthor is configured to true, it means Renovate will fetch the entire list of repository PRs instead of optimizing to fetch only those PRs which it created itself.
>You should only want to enable this if you are changing the bot account (e.g. from @old-bot to @new-bot) and want @new-bot to find and update any existing PRs created by @old-bot.
>It's recommended to revert this setting once that transition period is over and all old PRs are resolved.

I think it could be more usefull if we can filter MR by labels for example. All ours MR have a label 'renovate'.  
A new option like 'prFilterByLabels' would be great

**Reply by PhilipAbed** [reply]
I would love that but how do you suggest we get those labels? This is too much manual work to be maintained and no way that i can see to automate this

**Reply by rarkins** [reply]
Any solution shouldn't assume the existence of labels
